### PR TITLE
Fix issues related to the trader's gui 

### PR DESCRIPTION
**Pull request by korobya**

### CODE REVIEW

File: Systems/Dialogue/BehaviorConversable.cs [hunk 1]
```diff
@@ -210,7 +210,7 @@ public override void OnGameTick(float deltaTime)
                         sapi.Network.SendEntityPacket(player as IServerPlayer, entity.EntityId, CloseConvoPacketId);
                     } else
                     {
-                        (world.Api as ICoreClientAPI).Network.SendEntityPacket(entity.EntityId, CloseConvoPacketId);
```
Comment (korobya) [inline Systems/Dialogue/BehaviorConversable.cs:213]: will be called in https://github.com/anegostudios/vssurvivalmod/blob/afc94069d30e0285185f5bf2dd798fe1b99fc2eb/Systems/Dialogue/BehaviorConversable.cs#L288

File: Systems/Auction/Auction.cs [hunk 2]
```diff
@@ -325,8 +325,6 @@ private void Event_PlayerDisconnect(IServerPlayer byPlayer)
         {
             if (createAuctionSlotByPlayer.TryGetValue(byPlayer.PlayerUID, out var inv))
             {
-                inv.DropAll(byPlayer.Entity.Pos.XYZ);
```
Comment (korobya) [inline Systems/Auction/Auction.cs:328]: this is not necessary, as it is called when the inventory is closed

https://github.com/anegostudios/vssurvivalmod/blob/afc94069d30e0285185f5bf2dd798fe1b99fc2eb/Systems/Auction/Auction.cs#L346-L348

**Comment by korobya** [comment]
[anegostudios/VintageStory-Issues#2416](https://github.com/anegostudios/VintageStory-Issues/issues/2416) seems to be caused by the following

https://github.com/anegostudios/vssurvivalmod/blob/afc94069d30e0285185f5bf2dd798fe1b99fc2eb/Systems/Trading/EntityTrader.cs#L542-L547

https://github.com/anegostudios/vssurvivalmod/blob/afc94069d30e0285185f5bf2dd798fe1b99fc2eb/Systems/Trading/EntityTrader.cs#L40

this doesn't seem to have a value assigned to it anywhere

**Comment by tyronx** [comment]
Hey, can you please explain in greater detail what these changes are for?

**Comment by korobya** [comment]
[abf68a9](https://github.com/anegostudios/vssurvivalmod/pull/34/commits/abf68a9ccb5ef23d9ba51b1810d48f25cf2149b7) commit resolves the issue related with duplicated audit logs. I discussed this on the [#gamedev](https://discord.com/channels/302152934249070593/402173001178677268/1196706387716227082) channel in discord.

[f2e7987](https://github.com/anegostudios/vssurvivalmod/pull/34/commits/f2e7987c41010af9be0909d0554501683efed455) and [6885938](https://github.com/anegostudios/vssurvivalmod/pull/34/commits/68859382ad41bdabf5345d40f3600efc224a2a8b) commits resolve https://github.com/anegostudios/VintageStory-Issues/issues/3310 and https://github.com/anegostudios/VintageStory-Issues/issues/1045

It seems that Th3Dilli has fixed https://github.com/anegostudios/VintageStory-Issues/issues/3541 and https://github.com/anegostudios/VintageStory-Issues/issues/2416. And it seems that he didn't notice my PR, as he has just added additional code to handle these bugs. 

https://github.com/anegostudios/vssurvivalmod/blob/bc2a1d7313ffad7a59aa16a050c65cf7422a8772/Gui/GuiDialogTrader.cs#L521-L530
https://github.com/anegostudios/vssurvivalmod/blob/bc2a1d7313ffad7a59aa16a050c65cf7422a8772/Systems/Dialogue/GuiDialogueDialog.cs#L147-L156

The changes i suggest in the [316441d](https://github.com/anegostudios/vssurvivalmod/pull/34/commits/316441d565da0586907b86d07eda9daaa6765574) commit fix existing code that just doesn't work properly which is why https://github.com/anegostudios/VintageStory-Issues/issues/3541 was present.

In https://github.com/anegostudios/vssurvivalmod/pull/34#issuecomment-1919718865, I explain what caused the https://github.com/anegostudios/VintageStory-Issues/issues/2416. I didn't fix it.

If his approach to solving this problem is satisfactory, then I can undo [316441d](https://github.com/anegostudios/vssurvivalmod/pull/34/commits/316441d565da0586907b86d07eda9daaa6765574) commit, making it more convenient for you to merge my PR.

**Comment by tyronx** [comment]
Oh, I'll ask Dilli to take over then

**Comment by Th3Dilli** [comment]
thank you very much @korobya 

I did not see your PR initially and yeah I did implemented similar functionality when addressing some of the issues


I have now included your fixes and changed to use the existing code for closing trader trading and trader dialog GUI's

the public repos will be updated as soon as the version including your fixes is released

**Pull request closed without merging**